File: Algorithm/Programmers/Level2/solution(5).py
# N개의 최소공배수

def solution(arr):
    answer = 0
    
    def gcd (a,b): # 최대공약수 구하기
        if a > b:
            a, b = b, a
        while b:
            a, b = b, a%b
        return a
    
    def lcm (a,b): # 최소공배수 구하기
        return a*b//gcd(a,b)
    
    while True:
        if len(arr) == 1: # 길이가 1이 되면 비교할 수가 없으므로 return
            return arr[0]
        arr.append(lcm(arr.pop(), arr.pop())) # pop 시켜서 최소공배수 계속 update
        
    return answer

File: Algorithm/Programmers/Level2/solution(5).java
// N개의 최소공배수
// 순서대로 값의 최소공배수를 구해가면 됨
// 최소공배수 공식 = 두 수의 곱 // 최대공약수

class Solution {
    
    // 최대공약수 구하기
    public int gcd(int a, int b){
        while (a != 0 && b != 0){
            if (a > b)
                a = a % b;
            else
                b = b % a;
        }
        return a + b;
    }
    
    // 최소공배수 구하기
    public int lcm(int a, int b){
        return a * b / gcd(a, b);
    }
    
    public int solution(int[] arr) {
        int answer = arr[0];
        for (int i = 1; i < arr.length; i++)
            answer = lcm(answer, arr[i]);
        return answer;
    }
}